File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
import 'dart:math';

class DifferenceOfSquares {

  int squareOfSum(int num) {
    int sum = 0;
    for (var i = num; i > 0; i--) {
      sum += i;
    }
    return pow(sum, 2).toInt();
  }

  int sumOfSquares(int num) {
    int sum = 0;
    for (var i = num; i > 0; i--) {
      sum += pow(i, 2).toInt();
    }
    return sum;
  }

  int differenceOfSquares(int num) {
    return squareOfSum(num) - sumOfSquares(num);
  }
}

File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

import "math"

func SquareOfSum(n int) int {
	return int(math.Pow(float64(squareOfSum(n)), 2.0))
}

func squareOfSum(n int) int {
    if n == 1 {
        return n
    }
    return n + squareOfSum(n -1)
}

func SumOfSquares(n int) int {
	return sumOfSquares(n)
}

func sumOfSquares(n int) int {
    if n == 1 {
        return n
    }
    return int(math.Pow(float64(n), 2.0)) + sumOfSquares(n -1)
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
